teuthology/provision/fog.py

```diff
@@ -0,0 +1,302 @@
+        if verify:
+            resp.raise_for_status()
+        return resp
+
+    def get_host_data(self):
+        """
+        Locate the host we want to use, and return the FOG object which
+        represents it
+        :returns: A dict describing the host
+        """
+        resp = self.do_request(
+            '/host',
+            data=json.dumps(dict(name=self.shortname)),
+        )
+        obj = resp.json()
+        if obj['count'] == 0:
+            raise RuntimeError("Host %s not found!" % self.shortname)
+        if obj['count'] > 1:
+            raise RuntimeError(
+                "More than one host found for %s" % self.shortname)
+        return obj['hosts'][0]
+
+    def get_image_data(self):
+        """
+        Locate the image we want to use, and return the FOG object which
+        represents it
+        :returns: A dict describing the image
+        """
+        name = '_'.join([
+            self.remote.machine_type, self.os_type.lower(), self.os_version])
+        resp = self.do_request(
+            '/image',
+            data=json.dumps(dict(name=name)),
+        )
+        obj = resp.json()
+        if not obj['count']:
+            raise RuntimeError(
+                "Could not find an image for %s %s",
+                self.os_type,
+                self.os_version,
+            )
+        return obj['images'][0]
+
+    def set_image(self, host_id):
+        """
+        Tell FOG to use the proper image on the next deploy
+        :param host_id: The id of the host to deploy
+        """
+        image_data = self.get_image_data()
+        image_id = int(image_data['id'])
+        self.do_request(
+            '/host/%s' % host_id,
+            method='PUT',
+            data=json.dumps(dict(imageID=image_id)),
+        )
+
+    def schedule_deploy_task(self, host_id):
+        """
+        :param host_id: The id of the host to deploy
+        :returns: The id of the scheduled task
+        """
+        self.log.info(
+            "Scheduling deploy of %s %s",
+            self.os_type, self.os_version)
+        # First, we need to find the right tasktype ID
+        resp = self.do_request(
+            '/tasktype',
+            data=json.dumps(dict(name='deploy')),
+        )
+        tasktypes = [obj for obj in resp.json()['tasktypes']
+                     if obj['name'].lower() == 'deploy']
+        deploy_id = int(tasktypes[0]['id'])
+        # Next, schedule the task
+        resp = self.do_request(
+            '/host/%i/task' % host_id,
+            method='POST',
+            data='{"taskTypeID": %i}' % deploy_id,
+        )
+        host_tasks = self.get_deploy_tasks()
+        for task in host_tasks:
+            timestamp = task['createdTime']
+            time_delta = (
+                datetime.utcnow() - datetime.strptime(
+                    timestamp, self.timestamp_format)
+            ).total_seconds()
+            # FIXME yay magic
+            if time_delta < 5:
```

Comment:
some more additional comments here please, what is <5 :)

Reply:
Yeah - a more appropriate comment would be good

Reply:
for me it would be totally resolved with s/time_delta/time_delta_in_sec/